File: 1147-flip-columns-for-maximum-number-of-equal-rows/1147-flip-columns-for-maximum-number-of-equal-rows-BRUTE-FORCE.py
#analyse that after fliiping to optimum we realise the columns which are similar and complementary give the right answer!
class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        maxIdenticalRows=0
        for row in matrix:
            flipped=[1-x for x in row]
            count=0
            for compareRow in matrix:
                if compareRow==row or compareRow==flipped:
                    count+=1
            maxIdenticalRows=max(maxIdenticalRows,count)
        return maxIdenticalRows

File: 1147-flip-columns-for-maximum-number-of-equal-rows/1147-flip-columns-for-maximum-number-of-equal-rows-BRUTE-FORCE.java
class Solution {
    public int[] flipIt(int[] row){
        int[] flipped=new int[row.length];
        for(int i=0;i<row.length;i++){
            flipped[i]=1-row[i];
        }
        return flipped;
    }
    public int maxEqualRowsAfterFlips(int[][] matrix) {
        int maxIdenticalRows=0;
        for(int[] row:matrix){
            int[] flipped=flipIt(row);
            int count=0;
            for(int[] compareRow:matrix){
                if(Arrays.equals(compareRow,row)||Arrays.equals(compareRow,flipped)){
                    count++;
                }
            }
            maxIdenticalRows=Math.max(count,maxIdenticalRows);
        }
        return maxIdenticalRows;
    }
}
